File: api/app/application/services/status_service.py
from app.domain.models.health_status import HealthStatus
from app.domain.external.health_checker import HealthChecker
import asyncio


class StatusService:
    def __init__(self, health_checkers: list[HealthChecker]):
        self._health_checkers = health_checkers

    async def check_all(self) -> list[HealthStatus]:
        statuses = []
        results = await asyncio.gather(
            *[checker.check() for checker in self._health_checkers],
            return_exceptions=True
        )
        for res in results:
            if isinstance(res, Exception):
                statuses.append(
                    HealthStatus(
                        status="error",
                        service="未知服务",
                        detail="未知服务未响应",
                    )
                )
            else:
                statuses.append(res)
        return statuses

File: api-node/src/services/status.service.ts
import { getPrisma } from '../infrastructure/database/prisma';
import { getRedis } from '../infrastructure/redis/client';
import { HealthStatus } from '../domain/models';

export class StatusService {
  async checkAll(): Promise<HealthStatus[]> {
    const statuses: HealthStatus[] = [];

    try {
      await getPrisma().$queryRaw`SELECT 1`;
      statuses.push({ status: 'ok', service: 'postgres', detail: 'connected' });
    } catch (e) {
      statuses.push({ status: 'error', service: 'postgres', detail: (e as Error).message });
    }

    try {
      const redis = getRedis();
      if (redis.status !== 'ready') await redis.connect();
      await redis.ping();
      statuses.push({ status: 'ok', service: 'redis', detail: 'connected' });
    } catch (e) {
      statuses.push({ status: 'error', service: 'redis', detail: (e as Error).message });
    }

    return statuses;
  }
}

export const statusService = new StatusService();
